src/NServiceBus.Core/Transports/Msmq/MsmqFailureInfoStorage.cs

```diff
@@ -67,6 +67,11 @@ public bool TryGetFailureInfoForMessage(string messageId, out ProcessingFailureI
 
         public void ClearFailureInfoForMessage(string messageId)
         {
+            if (string.IsNullOrEmpty(messageId))
```

Comment:
At first @dbelcham and I were thinking it might be possible for `message` to be null at the call site, but looking at it now, I don't think this is possible.